### PR TITLE
Update to use dimod 0.10

### PR DESCRIPTION
**Pull request by arcondello**
~WIP: Currently I am getting a double free error~

### CODE REVIEW

File: dwave/preprocessing/include/dwave-preprocessing/posiform_info.hpp [hunk 1]
```diff
@@ -61,8 +61,8 @@ to the floating point based numbers as many biases may be flushed to zeroes.
 public:
   using coefficient_type = coefficient_t; // Must be a signed integral type.
   using bias_type = typename BQM::bias_type;
-  using quadratic_iterator_type = typename BQM::const_outvars_iterator;
-  using variable_type = typename BQM::variable_type;
+  using quadratic_iterator_type = typename BQM::const_neighborhood_iterator;
```
Comment (amahmudwave) [inline dwave/preprocessing/include/dwave-preprocessing/posiform_info.hpp:64]: I am not sure this code is compatible with the new type of iterator.

Comment (arcondello) [reply dwave/preprocessing/include/dwave-preprocessing/posiform_info.hpp:64]: Fair enough, just curious if the fix should be in dimod or here. What are the assumptions that are now violated?

Comment (amahmudwave) [reply dwave/preprocessing/include/dwave-preprocessing/posiform_info.hpp:64]: So I assume it traverses the whole matrix, not the lower or upper triangular.

Comment (arcondello) [reply dwave/preprocessing/include/dwave-preprocessing/posiform_info.hpp:64]: `BQM::const_neighborhood_iterator` does, it's still traverses the entire neighborhood of the given variable, assuming you're calling it vias `BQM::neighborhood`.

**Pull request merged**